src/compiler/parser.ts

```diff
@@ -6080,7 +6080,14 @@ namespace ts {
             // one out no matter what.
             let finallyBlock: Block | undefined;
             if (!catchClause || token() === SyntaxKind.FinallyKeyword) {
-                parseExpected(SyntaxKind.FinallyKeyword);
+                const catchOrFinallyExpectedMessage: DiagnosticMessage = {
+                    code: 1005,
+                    category: DiagnosticCategory.Error,
+                    key: "_0_expected_1005",
+                    message: "'catch' or 'finally' expected."
+                };

```

Comment:
I don't think it's quite right to do this here; diagnostic messages are generated from `diagnosticMessages.json` so they can be localized, so this probably should be going there, then referenced like other messages.

Reply:
Good point, thanks Jake. I just added a commit moving the new message to diagnosticMessages.json.